frappe/database/mariadb/database.py

```diff
@@ -114,18 +114,23 @@ def set_execution_timeout(self, seconds: int):
 		self.sql("set session max_statement_time = %s", int(seconds))
 
 	def get_connection_settings(self) -> dict:
+		assert (
+			self.password
+		), "without unix domain socket connection, mariadb db password is currently reqired"
```

Comment:
This assert already hints on how this prepares for better developer UX via passwordless development workflows in the local context.